### PR TITLE
Dw 101 upgradepopup

### PR DESCRIPTION
**Pull request by cbernat**
For monthly subscriber or demo enviroment, click on buy and the popup did not show.

image

image

- Removed hook for intl
- Add test to validate correct render

### CODE REVIEW

File: src/components/UpgradePlanForm/UpgradePlanForm.js [hunk 1]
```diff
@@ -11,12 +11,12 @@ const fieldNames = {
...
 
 class UpgradePlanForm extends React.Component {
-  constructor({ dependencies: { dopplerLegacyClient } }) {
+  constructor({ dependencies: { dopplerLegacyClient }, intl }) {
     super();
 
     /** @type { import('../../services/doppler-legacy-client').DopplerLegacyClient } */
     this.dopplerLegacyClient = dopplerLegacyClient;
-    this.intl = useIntl();
+    this.intl = intl;
```
Comment (fcoronelmakingsense) [inline src/components/UpgradePlanForm/UpgradePlanForm.js]: why can't use hook?

File: src/components/UpgradePlanForm/UpgradePlanForm.test.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import '@testing-library/jest-dom/extend-expect';
+import IntlProvider from '../../i18n/DopplerIntlProvider.double-with-ids-as-values';
+import UpgradePlanForm from './UpgradePlanForm';
+import { AppServicesProvider } from '../../services/pure-di';
+
+describe('Upgrade plan form component', () => {
+  afterEach(cleanup);
+
+  it('renders upgrade popup for monthly subscribers without errors', async () => {
+    const dopplerLegacyClientDouble = {
```
Comment (fcoronelmakingsense) [inline src/components/UpgradePlanForm/UpgradePlanForm.test.js]: You can add `Arrange, Act, Assert` comments for more clear

File: src/components/UpgradePlanForm/UpgradePlanForm.js (hunk 1 above)
Comment (andresmoschini) [reply src/components/UpgradePlanForm/UpgradePlanForm.js]: I have the same question, what is the rationale behind fixing the bug removing the hook?

Comment (cbernat) [reply src/components/UpgradePlanForm/UpgradePlanForm.js]: We cannot use hooks inside a class component, this was the error. It breaks the code.

Comment (cbernat) [reply src/components/UpgradePlanForm/UpgradePlanForm.js]: https://reactjs.org/docs/hooks-faq.html#should-i-use-hooks-classes-or-a-mix-of-both

**Pull request merged**